Add Temporal Cloud pricing page (#2134)
temporalio/documentation: docs-src/cloud-context/pricing.md

---
id: pricing
title: Temporal Cloud Pricing
sidebar_label: Pricing
description: "Temporal Cloud pricing information."
tags:
  - guide-context
---

> Temporal Cloud is a consumption-based service; you pay only for what you need when you need it. Pricing is flexible, transparent, and predictable, so you know your costs and never pay for unused components.

Reviewer: ```suggestion
Temporal Cloud is a consumption-based service; you pay only for what you need when you need it. Pricing is flexible, transparent, and predictable, so you know your costs and never pay for unused capacity.
```
Changed in source document.

This page introduces the elements of Temporal Cloud pricing so that you can estimate costs for your implementation. To gain a reliable estimate, [contact our team](https://pages.temporal.io/contact-us).

## Temporal Actions

Actions are the fundamental consumption pricing unit in Temporal Cloud. An Action in Temporal occurs as part of an execution of your Workflow. Each time you execute a Temporal Workflow (a Workflow Execution), the associated Actions are collected and ultimately represent the state and progress of your Temporal Application.

> For a deeper description of Actions, please refer to [this section](/cloud#action) of our documentation. You can also [reach out to our team](https://pages.temporal.io/contact-us) to get more information or to help size your number of Actions.

Reviewer: ```suggestion
For a deeper description of Actions, see [Action](/cloud#action) of our documentation. You can also [reach out to our team](https://pages.temporal.io/contact-us) to get more information or to help size your number of Actions.
```
- https://developers.google.com/style/word-list#please
- https://developers.google.com/style/link-text#write-link-text
- Referring to "our documentation" when using the documentation.

| 30,000 to 149,999                | $7.90                 | $237,000–$1,185,000 | ~57,000                |
| 150,000 or more                  | $5.90                 | Begins at $885,000  | n/a                    |

## Storage

An execution of a particular Workflow could exist for a few seconds, a day, month, or even forever. Temporal collects the Event History during this time and dispatches work when necessary. In this context, a Workflow Execution has only two states, open (active) or closed.

Storage costs are measured in gigabyte-hours (GBh) and include charges for active Workflows, "running" storage, and the long-term, "retained" storage of Event Histories of closed Workflows. These are measured per Namespace.

Running storage is a measure of the amount of storage used to store active Workflows. When the execution of a Workflow ends, Temporal Cloud stores Event History for a defined Retention Period, for historical use. This is retained storage. Typical uses include compliance, debugging, workload refresh, and business analytics. Both kinds of storage have fixed costs.

> | **Retained storage** | $0.00042 per GBh |
> | **Running storage** | $0.042 per GBh |

Reviewer: My previous idea was well intended but erroneous. Try this new new idea:
```suggestion
| **Storage** | **Cost per GBh** |
| ----------- | ---------------- |
| Retained    | $0.00042         |
| Running     | $0.042           |
```

> If you purchase Temporal Cloud credits (as outlined above), running storage costs are tiered and measured in gigabyte-hours.

Reviewer: ```suggestion
If you purchase Temporal Cloud credits (as outlined earlier), running storage costs are tiered and measured in gigabyte-hours.
```

| **Type**       | **Basic support**                                                                            | **Premium support**                                                                    |
| -------------- | -------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------- |
| Availability   | 0900–1700 Mon–Fri (Pacific Time)                                                             | 24×7                                                                                   |
| Response times | P0: 1 business hour<br/>P1: 4 business hours<br/>P2: 1 business days<br/>P3: 2 business days | P0: 30 minutes<br/>P1: 1 business hour<br/>P2: 4 business hours<br/>P3: 1 business day |
| Pricing        | Greater of $200 or 10% monthly usage (per month)                                             | Greater of $2,000 or 10% monthly activity (per month)                                  |

> | //             |                                                                                              |                                                                                        |

Reviewer: ```suggestion
```

## SSO and SAML

We offer single sign-on (SSO) integration using SAML at a monthly fixed fee based on the number of users registered in Temporal Cloud:

> | **Users** | **Cost per Month** |

Reviewer: ```suggestion
| **Users** | **Cost per month** |
```

---
id: pricing
title: Temporal Cloud Pricing
sidebar_label: Pricing
description: "Temporal Cloud pricing information."
tags:
  - guide-context
---

Temporal Cloud is a consumption-based service for which you pay for only what you need when you need it. Pricing is flexible, transparent, and predictable, so that you know your costs and never pay for unused components.

This document walks through the elements of Temporal Cloud pricing and is a guide to help you understand concepts so you can estimate costs for your implementation. To gain a reliable estimate, you should [engage with our team](https://pages.temporal.io/contact-us).

## Pricing Element: Temporal Actions

Actions are the fundamental consumption pricing unit in Temporal Cloud. An Action in Temporal is an event that occurs as part of an execution of your Workflow. Each time you execute a Temporal Workflow (a Workflow Execution), the associated Actions are collected and ultimately represent the state and progress of your Temporal Application.

> For a deeper description of Actions, please refer to [this section](LINK:https://docs.temporal.io/cloud#action) of our documentation. You can also [reach out to our team] (https://pages.temporal.io/contact-us) to get more information or to help size your number of Actions.

Reviewer 2: ```suggestion
For a deeper description of Actions, please refer to [this section](/cloud#action) of our documentation. You can also [reach out to our team](https://pages.temporal.io/contact-us) to get more information or to help size your number of Actions.
```

| 30,000 to 149,999                | $7.90                 | $237,000 - $1,185,000 | ~57,000                |
| > 150,000                        | $5.90                 | $885,000 ->           | –                      |

## Pricing Element: Storage

An execution of a particular Workflow could exist for a few seconds, a day, month, or even forever. Temporal collects the Event History during this time and dispatches work when necessary. In this context, a Workflow Execution has only two states, open (active) or closed.

Storage costs are measured in gigabyte-hours (GBh) and include charges for active Workflows, "running storage," and the long-term, "retained" storage of Event Histories of closed Workflows. These are measured per Namespace.

Running storage is a measure of the amount of storage used to store active Workflows. When the execution of a Workflow ends, Temporal Cloud stores Event History for a defined Retention Period, for historical use. This is retained storage. Typical use includes compliance, debugging, workload refresh, and business analytics. These both have fixed costs:

> | **Retained Storage** | $0.00042 per GBh |
> | -------------------- | ---------------- |
> | **Running Storage**  | $0.042 per GBh   |

Reviewer: ```suggestion
| **Retained storage** | $0.00042 per GBh |
| **Running storage**  | $0.042 per GBh   |
```

If you purchase Temporal Cloud credits (as outlined above), running storage costs are tiered and measured in gigabyte-hours.

> | **Running Storage** | **Cost per GBh** |

Reviewer: ```suggestion
| **Running storage** | **Cost per GBh** |
```

| ------------------- | ---------------- |
| < 10                | $0.042           |
| 10 to 39            | $0.031           |
| 40 to 119           | $0.023           |
| 120 to 499          | $0.018           |
| 500 to 1999         | $0.013           |
| > 2000              | $0.010           |

## Pricing Element: Support

> With a subscription to Temporal Cloud you also gain access to our support organization made up of developer success engineers and solution architects. Our team of experts assist with a range of work streams from Workflow design reviews to setting up observability to break/fix support with an agreed-upon set of SLAs.

Reviewer: ```suggestion
With a subscription to Temporal Cloud, you also gain access to our support organization of developer success engineers and solution architects. Our experts assist with a range of work streams from Workflow design reviews to setting up observability to break/fix support with an agreed-upon set of SLAs.
```

The Temporal Developer Success team is composed of open source contributors and is backed by the main contributors to the open source Temporal project. They provide deep knowledge of how Temporal works and how you can optimize your deployment.

They also investigate your Workflows to optimize their environments and possibly reduce costs associated with Actions and storage. They ensure your instance is performance tuned and help with other ongoing maintenance, like upgrades of the Temporal software and maintenance of the platform.

> If an issue occurs, the team provides support through email, phone, forums, knowledge base and an optional dedicated slack channel. Temporal offers two levels of support defined by their availability and SLAs in the following table.

Reviewer: ```suggestion
If an issue occurs, the team provides support through email, phone, forums, knowledge base, and an optional dedicated Slack channel. Temporal offers two levels of support defined by their availability and SLAs.
```

> | **Type**       | **Basic Support**                                                             | **Premium Support**                                                     |

Reviewer: ```suggestion
| **Type**       | **Basic support**                                                             | **Premium support**                                                     |
```

| -------------- | ----------------------------------------------------------------------------- | ----------------------------------------------------------------------- |

> | Availability   | 0900–1700 Mon–Fri (GMT-9)                                                     | 24×7                                                                    |

Reviewer: ```suggestion
| Availability   | 0900–1700 Mon–Fri (Pacific Time)                                                     | 24×7                                                                    |
```
GMT-9 is Alaska time.

> | Response times | P0: 1 business hourP1: 4 business hoursP2: 1 business daysP3: 2 business days | P0: 30 minutesP1: 1 business hourP2: 4 business hoursP3: 1 business day |

Reviewer: ```suggestion
| Response times | P0: 1 business hour<br>P1: 4 business hours<br>P2: 1 business days<br>P3: 2 business days | P0: 30 minutes<br>P1: 1 business hour<br>P2: 4 business hours<br>P3: 1 business day |
```

> | Response times | P0: 1 business hourP1: 4 business hoursP2: 1 business daysP3: 2 business days | P0: 30 minutesP1: 1 business hourP2: 4 business hoursP3: 1 business day |

Reviewer 2: ```suggestion
| Response times | P0: 1 business hour, P1: 4 business hours, P2: 1 business days, P3: 2 business days | P0: 30 minutes, P1: 1 business hour, P2: 4 business hours, P3: 1 business day |
```

| Pricing        | Greater of $200 or 10% monthly usage (per month)                              | Greater of $2,000 or 10% monthly activity (per month)                   |

## Pricing Element: SSO and SAML

We offer single sign-on (SSO) integration using SAML at a monthly fixed fee based on the number of users registered in Temporal Cloud:

| **Users** | **Cost per Month** |
| --------- | ------------------ |
| 0 to 25   | $200               |
| 26 to 50  | $300               |
| 51+       | $500               |

## Pricing Estimates

> Temporal Cloud employs a consumption-based pricing model that's based on storage and execution, factors that vary from one Workflow to the next. You can estimate the cost of a specific Workflow by running it at a low volume and then using its storage and compute measurements to project your production scale cost. Our team is happy to [help you estimate the cost](https://pages.temporal.io/contact-us) for your specific workloads.

Reviewer: ```suggestion
Temporal Cloud employs a consumption-based pricing model that's based on storage and execution, factors that vary from one Workflow to the next. You can estimate the cost of a specific Workflow by running it at a low volume and then using its storage and compute measurements to project your production-scale cost. Our team is happy to [help you estimate the cost](https://pages.temporal.io/contact-us) for your specific workloads.
```

> ## Temporal Pricing FAQ

Reviewer: The contents of this section also appear in the proposed Cloud FAQ section (EDU-869). Remove from this page now and link to the FAQ when it's ready? Or remove from FAQ page (which currently leaves only two FAQs)?

```suggestion
## Temporal pricing FAQ
```

> Q: What’s the minimum fee to run Temporal Cloud?

Reviewer: ```suggestion
Q: What’s the minimum cost to run Temporal Cloud?
```

> A: The Temporal Cloud service is consumption based. You pay only for what you need with no minimum. Basic Support has a minimum monthly fee of $200 per month.

Reviewer: ```suggestion
A: The Temporal Cloud service is consumption based. You pay only for what you need with no minimum. Basic support has a minimum monthly fee of $200 per month.
```
